[HUDI-973] RemoteHoodieTableFileSystemView supports non-partitioned table queries

cxzl25 opened the pull request:
## What is the purpose of the pull request

When hoodie.embed.timeline.server = true, the written table is a non-partitioned table, will get an exception.
```java
io.javalin.BadRequestResponse: Query parameter 'partition' with value '' cannot be null or empty
	at io.javalin.validation.TypedValidator.getOrThrow(Validator.kt:25)
	at org.apache.hudi.timeline.service.FileSystemViewHandler.lambda$registerDataFilesAPI$3(FileSystemViewHandler.java:172)
```
Because api checks whether the value of partition is null or empty.
```java
ctx.validatedQueryParam(RemoteHoodieTableFileSystemView.PARTITION_PARAM).getOrThrow()
```

## Brief change log
  RemoteHoodieTableFileSystemView supports non-partitioned table queries


## Verify this pull request
  Manually verified the change by running a job locally

## Committer checklist

 - [ ] Has a corresponding JIRA in PR title & commit
 
 - [ ] Commit message is descriptive of the change
 
 - [ ] CI is green

 - [ ] Necessary doc changes done or have another open PR
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

cxzl25 commented:
Enable timeline-server by default(https://github.com/apache/hudi/pull/1634), may use the previous version to write non-partitioned table operations, throw an exception, and switch to AbstractTableFileSystemView.

bvaradar commented:
@cxzl25 : This is good catch. Thanks for fixing it. We handle by falling back to local file-system view but this is inefficient. This shows a gap in testing non-partitioned table in TestHoodieTableFileSystemView. We need to add a test case there. 

@nsivabalan :  this would be good to have in 0.5.3 release.

bvaradar commented:
@cxzl25 : Have opened a jira https://jira.apache.org/jira/browse/HUDI-975 to add unit-test in general for non-partitioned table across all of file-system view implementations. Would you be interested in taking that as a follow-up ?

I will go ahead and merge this code as I think this should go into 0.5.3.  cc @nsivabalan

cxzl25 commented:
@bvaradar :  I chose a method in TestHoodieTableFileSystemView, added a partition path parameter, and added a unit test, but I am not sure whether all unit tests need to be modified like this. (https://github.com/apache/hudi/pull/1692)
